iphone/Classes/CMHelper.m

```diff
@@ -135,18 +135,18 @@ +(NSDictionary*)dictionaryFromMotionActivity:(CMMotionActivity*)motionActivity
 +(NSDictionary*)dictionaryFromPedometerData:(CMPedometerData*)pedometerData
 {
     NSMutableDictionary *dict = [NSMutableDictionary dictionaryWithDictionary:@{
-        @"startDate": (!pedometerData) ? [NSNull null] : [TiUtils UTCDateForDate:pedometerData.startDate],
-        @"endDate": (!pedometerData) ? [NSNull null] : [TiUtils UTCDateForDate:pedometerData.endDate],
-        @"numberOfSteps": (!pedometerData) ? [NSNull null] : NUMINT(pedometerData.numberOfSteps),
-        @"distance": (!pedometerData) ? [NSNull null] : NUMINT(pedometerData.distance),
-        @"floorsAscended": (!pedometerData) ? [NSNull null] : NUMINT(pedometerData.floorsAscended),
-        @"floorsDescended": (!pedometerData) ? [NSNull null] : NUMINT(pedometerData.floorsDescended)        
+        @"startDate": (!pedometerData.startDate) ? [NSNull null] : [TiUtils UTCDateForDate:pedometerData.startDate],
+        @"endDate": (!pedometerData.endDate) ? [NSNull null] : [TiUtils UTCDateForDate:pedometerData.endDate],
+        @"numberOfSteps": (!pedometerData.numberOfSteps) ? [NSNull null] : pedometerData.numberOfSteps,
+        @"distance": (!pedometerData.distance) ? [NSNull null] : pedometerData.distance,
+        @"floorsAscended": (!pedometerData.floorsAscended) ? [NSNull null] : pedometerData.floorsAscended,
+        @"floorsDescended": (!pedometerData.floorsDescended) ? [NSNull null] : pedometerData.floorsDescended
```

Comment:
In code I have made following change  -
 "NUMINT(pedometerData.numberOfSteps)" changed to "pedometerData.numberOfSteps" also.
Reason was "numberOfSteps" property was already NSNumber, but we were forcefully changing it to NSNumber assuming it is Int.  

 Reason why it was giving wrong result -
When we convert any NSNumber object (assuming it as Int) to NSNumber , it will take address of Object rather value . And it will convert its address value to NSNumber object .  So it was giving wrong value . 

  Please let me know if you have any question.